tenpy/simulations/time_evolution.py

```diff
@@ -74,3 +80,347 @@ def final_measurements(self):
+        if 'operator_t' and 'operator_t0' and 'final_time' not in self.options.keys():
+            raise KeyError("`operator_t`, `operator_t0` and a `final_time` must be supplied")
+        self.operator_t = self.options['operator_t']
+        # generate info for operator before time evolution as subconfig
+        self.operator_t0_config = self.options.subconfig('operator_t0')
+        self.operator_t0 = None  # read out config later, since defaults depend on model parameters
+        self.addJW = self.options.get('addJW', False)
+        # for resuming simulation from checkpoint # this is provided in super().__init__
+        # TODO: How to ensure resuming from checkpoint works, when evolve_bra is True ?
+        resume_data = self.results.get("resume_data", None)
+        if resume_data:
+            if 'psi_ground_state' in self.results['simulation_parameters'].keys():
+                self.psi_ground_state = self.results['simulation_parameters']['psi_ground_state']
+
+    @classmethod
+    def from_gs_search(cls, filename, sim_params, **kwargs):
+        """Initialize an instance of a :class:`SpectralSimulation` from
+        a finished run of :class:`GroundStateSearch`. This simply fetches
+        the relevant parameters ('model_params', 'psi')
+
+        Parameters
+        ----------
+        filename : str or dict
+            The filename of the ground state search output to be loaded.
+            Alternatively the results as dictionary.
+        sim_params : dict
+            The necessary simulation parameters, it is necessary to specify final_time (inherited from
+            :class:`RealTimeEvolution`). The parameters of the spectral simulation should also be given similar
+            to the example params in the :class:`SpectralSimulation`.
+        **kwargs :
+            Further keyword arguments given to :meth:`__init__` of the class :class:`SpectralSimulation`.
+        """
+        return cls(options=sim_params, gs_data=filename, **kwargs)
+
+    def init_state(self):
+        # make sure state is not reinitialized if psi and psi_ground_state are given
+        if not hasattr(self, 'psi_ground_state'):
+            gs_data = self.gs_data
+            if gs_data is not None:
+                if isinstance(gs_data, MPS):
+                    self.psi_ground_state = gs_data
+                else:
+                    self.psi_ground_state = gs_data['psi']
+                delattr(self, 'gs_data')  # possibly free memory
+            else:
+                self.logger.warning("No ground state data is supplied, calling the initial state builder on\
+                                     SpectralSimulation class. You probably want to supply a ground state")
+
+                super().init_state()  # this sets self.psi from init_state_builder (should be avoided)
+                self.psi_ground_state = self.psi.copy()
+                delattr(self, 'psi')  # free memory
```

Comment:
This will definitely need a rework (as the logic is over-complicated). 
However, the point is that if `psi` is set manually that should translate to setting the ground-state.
And psi needs to be removed (intermediately) in order to be compatible with checkpointing (otherwise we might apply the initial operator to a time evolved checkpoint again). Alternatively I could overwrite `resume_run` in a way that `init_state` is not called. But maybe it's better to fix the structure in a later version?